extras/test_hull_performace.cpp

```diff
@@ -0,0 +1,175 @@
+  PrintVolArea(input, hullManifold);
+  std::chrono::duration<double> elapsed = end - start;
+  std::cout << elapsed.count() << " sec";
+}
+
+void PrintCGAL(Manifold input, Manifold (Manifold::*hull_func)() const) {
+  TriangleMesh cgalInput;
+  manifoldToCGALSurfaceMesh(input, cgalInput);
+  std::vector<Point> points;
+  for (const auto &vert : cgalInput.vertices()) {
+    points.push_back(cgalInput.point(vert));
+  }
+  // Convex Hull
+  TriangleMesh cgalHull;
+  auto start = std::chrono::high_resolution_clock::now();
+  CGAL::convex_hull_3(points.begin(), points.end(), cgalHull);
+  auto end = std::chrono::high_resolution_clock::now();
+  PrintVolAreaCGAL(cgalInput, cgalHull);
+  std::chrono::duration<double> elapsed = end - start;
+  std::cout << elapsed.count() << " sec";
+}
+
+// Constructs a Menger Sponge, and tests the convex hull implementation on it
+// (you can pass the specific hull implementation to be tested). Comparing the
+// volume and surface area with CGAL implementation, for various values of
+// rotation
+void MengerTestHull(Manifold (Manifold::*hull_func)() const, float rx, float ry,
+                    float rz, char *implementation) {
+  Manifold sponge = MengerSponge(4);
+  sponge = sponge.Rotate(rx, ry, rz);
+  if (!strcmp(implementation, "Hull"))
+    PrintManifold(sponge, hull_func);
+  else if (!strcmp(implementation, "CGAL"))
+    PrintCGAL(sponge, hull_func);
+}
+
+// Constructs a high quality sphere, and tests the convex hull implementation on
+// it (you can pass the specific hull implementation to be tested). Comparing
+// the volume and surface area with CGAL implementation
+void SphereTestHull(Manifold (Manifold::*hull_func)() const,
+                    char *implementation) {
+  Manifold sphere = Manifold::Sphere(1, 6000);
+  sphere = sphere.Translate(glm::vec3(0.5));
+  if (!strcmp(implementation, "Hull"))
+    PrintManifold(sphere, hull_func);
+  else if (!strcmp(implementation, "CGAL"))
+    PrintCGAL(sphere, hull_func);
+}
+
+int main(int argc, char **argv) {
+  if (argc < 4) {
+    std::cout << "Usage: ./test_hull_performance <Test (Sphere/Menger/Input)> "
+                 "<Implementation (Hull/Hull_CGAL)> <Print Header (1/0)> "
+                 "<Input Mesh (filename)> "
+              << std::endl;
+    return 0;
+  }
+  if (!strcmp(argv[3], "1"))
+    std::cout
+        << "VolManifold,VolHull,AreaManifold,AreaHull,ManifoldTri,HullTri,Time"
+        << std::endl;
+  if (!strcmp(argv[1], "Sphere"))
+    SphereTestHull(&Manifold::Hull, argv[2]);
+  else if (!strcmp(argv[1], "Menger"))
+    MengerTestHull(&Manifold::Hull, 1, 2, 3, argv[2]);
+  else if (!strcmp(argv[1], "Input")) {
+    auto inputMesh = ImportMesh(argv[4], 1);
+    Manifold inputManifold = Manifold(inputMesh);
+    if (!strcmp(argv[2], "Hull"))
+      PrintManifold(inputManifold, &Manifold::Hull);
+    else if (!strcmp(argv[2], "Hull_CGAL"))
+      PrintCGAL(inputManifold, &Manifold::Hull);
```

Comment:
It looks like you're always using `Manifold::Hull()` - do we still need this input?

Reply:
Oh right, I forgot that we didn't need input of the hull function there, as we're using CGAL, I'll make that change, thanks for pointing it out. And the reason I am taking this input, is that we can run a comparison between various implementations (or for the bare minimum at least with CGAL).